### PR TITLE
feat(ui): Refactor FormField --> Field

### PR DESCRIPTION
**Pull request by billyvg**
This refactors `FormField` into more of a form state handling component. The presentational component is now `Field`.

- Adds a storybook story
- Use Tooltip component

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/views/settings/organization/general/organizationGeneralSettingsView.jsx [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import {browserHistory} from 'react-router';
 import PropTypes from 'prop-types';
 import React from 'react';
-import createReactClass from 'create-react-class';
 import Reflux from 'reflux';
+import createReactClass from 'create-react-class';
```
Comment (dcramer) [inline src/sentry/static/sentry/app/views/settings/organization/general/organizationGeneralSettingsView.jsx]: looks like this page doesn't have jest tests? are those hard to add for the new components (e.g. just a snapshot that has mocked api for rendering)?

Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/organization/general/organizationGeneralSettingsView.jsx]: Yeah I'll add them in this PR https://github.com/getsentry/sentry/pull/7303/files

File: src/sentry/static/sentry/app/views/settings/components/forms/field/fieldControl.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import {Box, Flex} from 'grid-emotion';
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+
+import FieldControlState from './fieldControlState';
+import Tooltip from '../../../../../components/tooltip';
+
+// This wraps Control + ControlError message
+// * can NOT be a flex box have because of position: absolute on "control error message"
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/views/settings/components/forms/field/fieldControl.jsx]: small typo here "have because"?

**Review by MaxBittker — APPROVED** [review]
refactor looks good to me as far as I can tell 👍

**Pull request merged**